### PR TITLE
[iOS] Introducing BluetoothConnection Framework

### PR DESCRIPTION
**Pull request by vladcorneci**
Extract and refactor the BLE transport code from GoldenGate bindings into a new
framework: `BluetoothConnection`.

Separate the connection establishing part in two steps. First, a generic BLE
Connector i.e. `BluetoothConnector` connects to a peripheral and discovers its
services and characteristics. Then, the connection gets mapped to a
`ConnectionType` by a `ConnectionResolver`. Both, the `ConnectionType` and the
`ConnectionResolver` are defined by the consumer.

Some generic UI code (e.g. PeerViewController) was also moved into the
`BluetoothConnection` framework since it can be reused by other 3rd party
applications to display BLE related information.

### CODE REVIEW

File: platform/apple/apps/host/Source/ComponentBase.swift [hunk 1]
```diff
@@ -223,7 +226,7 @@ class ComponentBase {
             elementConfigurationProvider: self
         )
     }
-
+
```
Comment (vladcorneci) [inline platform/apple/apps/host/Source/ComponentBase.swift:229]: nit: whitespaces

File: platform/apple/bindings/GoldenGate/Bridging/Spec/GGErrorTypeSpec.swift [hunk 2]
```diff
@@ -26,7 +26,7 @@ class GGErrorTypeSpec: QuickSpec {
                 expect(error.errorUserInfo).to(beEmpty())
             }
         }
-
+
```
Comment (vladcorneci) [inline platform/apple/bindings/GoldenGate/Bridging/Spec/GGErrorTypeSpec.swift:29]: nit: Whitespace

File: platform/apple/apps/host/GoldenGate-Host-iOS/HostVersion.swift [hunk 3]
```diff
@@ -0,0 +1,146 @@
+//  Copyright 2017-2020 Fitbit, Inc
+//  SPDX-License-Identifier: Apache-2.0
+//
+//  HostVersion.swift
```
Comment (radumvlad) [inline platform/apple/apps/host/GoldenGate-Host-iOS/HostVersion.swift:4]: is this used?

File: platform/apple/apps/host/GoldenGate-Host-macOS/Info.plist [hunk 4]
```diff
@@ -22,11 +22,13 @@
 	<string>1</string>
 	<key>LSMinimumSystemVersion</key>
 	<string>$(MACOSX_DEPLOYMENT_TARGET)</string>
+	<key>NSBluetoothAlwaysUsageDescription</key>
+	<string>We need to communicate with your GoldenGate device.</string>
 	<key>NSHumanReadableCopyright</key>
-	<string>Copyright © 2017-2020 Fitbit, Inc. All rights reserved.</string>
-	<key>NSPrincipalClass</key>
-	<string>NSApplication</string>
+	<string>Copyright © 2017 Fitbit. All rights reserved.</string>
```
Comment (radumvlad) [inline platform/apple/apps/host/GoldenGate-Host-macOS/Info.plist:28]: perhaps revert?

File: platform/apple/apps/host/GoldenGateHost.xcodeproj/xcshareddata/xcschemes/GoldenGateHost-RemoteTesting-iOS.xcscheme [hunk 5]
```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (radumvlad) [inline platform/apple/apps/host/GoldenGateHost.xcodeproj/xcshareddata/xcschemes/GoldenGateHost-RemoteTesting-iOS.xcscheme:1]: remove this?

File: platform/apple/apps/host/GoldenGate-Host-iOS/HostVersion.swift (hunk 3 above)
Comment (vladcorneci) [reply platform/apple/apps/host/GoldenGate-Host-iOS/HostVersion.swift:4]: - [ ] Remove `HostVersion.swift`

File: platform/apple/apps/host/GoldenGate-Host-macOS/Info.plist (hunk 4 above)
Comment (vladcorneci) [reply platform/apple/apps/host/GoldenGate-Host-macOS/Info.plist:28]: - [ ] revert

File: platform/apple/bindings/GoldenGate/Tests/ExampleSpec.swift [hunk 6]
```diff
@@ -1,10 +1,9 @@
-//  Copyright 2017-2020 Fitbit, Inc
-//  SPDX-License-Identifier: Apache-2.0
 //
 //  ExampleSpec.swift
 //  GoldenGate
 //
 //  Created by Sylvain Rebaud on 11/27/17.
+//  Copyright © 2017 Fitbit. All rights reserved.
```
Comment (radumvlad) [inline platform/apple/bindings/GoldenGate/Tests/ExampleSpec.swift:6]: SPDX-License-Identifier: Apache-2.0?

File: platform/apple/bindings/GoldenGate/UI/BlastingPlaygroundViewController.swift [hunk 7]
```diff
@@ -0,0 +1,192 @@
+//  Copyright 2017-2020 Fitbit, Inc
+//  SPDX-License-Identifier: Apache-2.0
+//
+//  BlastingPlaygroundViewController.swift
+//  GoldenGate
+//
+//  Created by Emanuel Jarnea on 27/07/2020.
+//  Copyright © 2020 Fitbit. All rights reserved.
```
Comment (radumvlad) [inline platform/apple/bindings/GoldenGate/UI/BlastingPlaygroundViewController.swift:8]: SPDX-License-Identifier: Apache-2.0?

File: platform/apple/connection/BluetoothConnection/BluetoothConnection.h [hunk 8]
```diff
@@ -0,0 +1,18 @@
+//
+//  BluetoothConnection.h
+//  BluetoothConnection
+//
+//  Created by Vlad Corneci on 05/06/2020.
+//  Copyright © 2020 Fitbit. All rights reserved.
```
Comment (radumvlad) [inline platform/apple/connection/BluetoothConnection/BluetoothConnection.h:6]: SPDX-License-Identifier: Apache-2.0

File: platform/apple/connection/BluetoothConnectionObjC/Utils/CBUUID+ArbitraryData.h [hunk 9]
```diff
@@ -0,0 +1,30 @@
+//
```
Comment (radumvlad) [inline platform/apple/connection/BluetoothConnectionObjC/Utils/CBUUID+ArbitraryData.h:1]: SPDX-License-Identifier: Apache-2.0

File: platform/apple/bindings/GoldenGate/Tests/ExampleSpec.swift (hunk 6 above)
Comment (radumvlad) [reply platform/apple/bindings/GoldenGate/Tests/ExampleSpec.swift:6]: Or should this be removed?

Comment (vladcorneci) [reply platform/apple/bindings/GoldenGate/Tests/ExampleSpec.swift:6]: - [ ] Delete

**Review by radumvlad — APPROVED** [review]

**Review by remusleonte — APPROVED** [review]

**Pull request merged**